renku/api/datasets.py

```diff
@@ -352,69 +363,102 @@ def _add_from_git(self, dataset, path, url, target, **kwargs):
                 submodule_path.relative_to(self.path).as_posix()
             ])
 
-        src = submodule_path / (target or '')
+        target = target or ''
+
+        src = submodule_path / target
 
-        if target and relative_to:
+        if relative_to:
             relative_to = Path(relative_to)
             if relative_to.is_absolute():
                 assert u.scheme in {
                     '', 'file'
                 }, 'Only relative paths can be used with URLs.'
-                target = (Path(url).resolve().absolute() / target).relative_to(
-                    relative_to.resolve()
+                relative_to = relative_to.resolve().relative_to(
+                    Path(url).resolve().absolute()
                 )
+
+            if target:
+                if str(target) != str(relative_to):
```

Comment:
Merge them together

```
if target and str(target) != str(relative_to)
```

or set the target default value first and then make the second check. We don't need 2 ifs here.

Also, comparing strings here is probably not the best way of going about this since they can be symbolic paths.